.gitignore

```diff
@@ -5,7 +5,6 @@
 coverage
 docs/gen
 node_modules/*
-package-lock.json
```

Comment:
I do not think it is a good idea to include the package-lock.json, it makes it harder for users to upgrade the dependencies.

Reply:
I'd like us to be using them, as it's best practice to: https://docs.npmjs.com/files/package-lock.json

We needed this PR because we were pulling in "^0.4.0" of x509, and therefore we ran into an issue with 0.4.5, and our builds stopped passing. I think a package-lock.json helps us identify this kind of issue going forward